src/main/java/calculator/Calculator.java

```diff
@@ -0,0 +1,88 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class Calculator implements Function<String, Integer> {
+    private static final int NULL_OR_EMPTY_RESULT = 0;
+    private static final String DEFAULT_DELIMITER = "[,:]";
+    private static final String CHECK_CUSTOM_DELIMITER_USED = "//";
+    private static final String CUSTOM_DELIMITER_REGEX = "//(.*)\n(.*)";
+
+    @Override
+    public Integer apply(String input) {
+        if (isNullOrEmpty(input)) {
+            return NULL_OR_EMPTY_RESULT;
+        }
+
+        if (input.startsWith(CHECK_CUSTOM_DELIMITER_USED)) {
+            return applyCustomDelimiter(input);
+        }
+
+        return applyDefaultDelimiter(input);
+    }
+
+    private Integer applyDefaultDelimiter(String input) {
+        final List<String> numbers = Arrays.stream(input.trim().split(DEFAULT_DELIMITER)).collect(Collectors.toList());
+        return numbers.stream()
+                .mapToInt(this::convertStringToInteger)
+                .sum();
```

Comment:
```suggestion
        return Arrays.stream(input.trim().split(DEFAULT_DELIMITER))
                .mapToInt(this::convertStringToInteger)
                .sum();
```

이 부분은 이렇게 코드를 줄여 볼 수 있을 것 같아요.